libraries/browser-tracker-core/src/tracker/out_queue.ts

```diff
@@ -333,25 +337,34 @@ export function OutQueueManager(
         executingQueue = false;
       }, connectionTimeout);
 
-      // The events (`numberToSend` of them), have been sent, so we remove them from the outQueue
-      // We also call executeQueue() again, to let executeQueue() check if we should keep running through the queue
-      const onPostSuccess = (numberToSend: number): void => {
+      const removeEventsFromQueue = (numberToSend: number): void => {
         for (let deleteCount = 0; deleteCount < numberToSend; deleteCount++) {
           outQueue.shift();
         }
         if (useLocalStorage) {
           attemptWriteLocalStorage(queueName, JSON.stringify(outQueue.slice(0, maxLocalStorageQueueSize)));
         }
+      };
+
+      // The events (`numberToSend` of them), have been sent, so we remove them from the outQueue
+      // We also call executeQueue() again, to let executeQueue() check if we should keep running through the queue
+      const onPostSuccess = (numberToSend: number): void => {
+        removeEventsFromQueue(numberToSend);
         executeQueue();
       };
 
       xhr.onreadystatechange = function () {
-        if (xhr.readyState === 4 && xhr.status >= 200 && xhr.status < 400) {
+        if (xhr.readyState === 4 && xhr.status >= 200) {
           clearTimeout(xhrTimeout);
-          onPostSuccess(numberToSend);
-        } else if (xhr.readyState === 4 && xhr.status >= 400) {
-          clearTimeout(xhrTimeout);
-          executingQueue = false;
+          if (xhr.status < 300) {
+            onPostSuccess(numberToSend);
+          } else {
+            if (!shouldRetryForStatusCode(xhr.status)) {
+              LOG.error(`Collector responded with status code ${xhr.status}, sending events won't be retried.`);
```

Comment:
I'm always trying to shorten strings where possible, always chasing that filesize goal, maybe this could just be `Status ${xhr.status}, will not retry`. All logs are prepended with `Snowplow:` so that should be obvious.